Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/SpatialCoordinateSystemManager.cs

```diff
@@ -53,188 +36,265 @@ public class SpatialCoordinateSystemManager : Singleton<SpatialCoordinateSystemM
         {
-            if (participants.ContainsKey(endpoint))
+            if (localizers.ContainsKey(localizer.SpatialLocalizerId))
             {
-                Debug.LogWarning("SpatialCoordinateSystemParticipant connected that already existed");
+                Debug.LogError($"Cannot register multiple SpatialLocalizers with the same ID {localizer.SpatialLocalizerId}");
                 return;
             }
 
-            if (spectatorView.Role == Role.Spectator)
+            localizers.Add(localizer.SpatialLocalizerId, localizer);
+        }
+
+        public void UnregisterSpatialLocalizer(ISpatialLocalizer localizer)
+        {
+            if (!localizers.Remove(localizer.SpatialLocalizerId))
             {
-                if (participants.Count > 0 &&
-                    !participants.ContainsKey(endpoint))
-                {
-                    Debug.LogWarning("A second SpatialCoordinateSystemParticipant connected while the device was running as a spectator. This is an unexpected scenario.");
-                    return;
-                }
+                Debug.LogError($"Attempted to unregister SpatialLocalizer with ID {localizer.SpatialLocalizerId} that was not registered.");
             }
+        }
 
-            DebugLog($"Creating new SpatialCoordinateSystemParticipant, Role: {spectatorView.Role}, IPAddress: {endpoint.Address}, SceneRoot: {transformedGameObject}, DebugLogging: {debugLogging}");
-            var actAsHost = spectatorView.Role == Role.User;
-            var participant = new SpatialCoordinateSystemParticipant(actAsHost, endpoint, WriteAndSendMessage, CreateDebugVisualParent, debugLogging, showDebugVisuals, debugVisual, debugVisualScale);
-            participants[endpoint] = participant;
-            if (spatialLocalizer != null)
+        public void RegisterNetworkManager(INetworkManager networkManager)
+        {
+            if (!networkManagers.Add(networkManager))
             {
-                DebugLog($"Localizing SpatialCoordinateSystemParticipant: {endpoint.Address}");
-                participant.LocalizeAsync(spatialLocalizer).FireAndForget();
+                Debug.LogError($"Attempted to register the same network manager multiple times");
+                return;
             }
-            else
+
+            RegisterEvents(networkManager);
+        }
+
+        public void UnregisterNetworkManager(INetworkManager networkManager)
+        {
+            if (!networkManagers.Remove(networkManager))
             {
-                Debug.LogWarning("Spatial localizer not specified for SpatialCoordinateSystemManager");
+                Debug.LogError($"Attempted to unregister a network manager that was not registered");
+                return;
             }
+
+            UnregisterEvents(networkManager);
         }
 
-        public void OnDisconnected(SocketEndpoint endpoint)
+        public void RestorePersistentSharedCoordinate(SocketEndpoint socketEndpoint, string sharedCoordinateName)
```

Comment:
>RestorePersistentSharedCoordinate [](start = 20, length = 33)

nit: add this when its used compared to now.

Reply:
Removed this for now